src/js/jsx/shared/Dialog.jsx

```diff
@@ -98,7 +98,7 @@ define(function (require, exports, module) {
         },
 
         componentWillMount: function () {
-            this.getFlux().actions.dialog.registerDialog(this.props.id, this._getDismissalPolicy());
+            this.getFlux().store("dialog").registerDialog(this.props.id, this._getDismissalPolicy());
```

Comment:
I have no alternate suggestion I guess.  I find it a little disconcerting that the action queue adds enough overhead such that your "direct" solution is noticeably faster.

In this particular case, is it the DE-registering that is gumming up the works?  (otherwise, why does the dialog sub-system slow down the marquee select?)  If so... is there a way to make that completely asynchronous so that the de-registration does not block marquee selection.  Part of me thinks we don't need to de-register them at all... but I suppose that could result in extra memory usage over time.

I'm just saying words at this point.  This is certainly not enough to block this PR ... but something still seems off to me.